Building qztest with qt4 fails on Linux

When building against qt4 on KDE Neon/Ubuntu 18.04 (build configured via qmake-qt4), the build fails with the following log:

> 
> n file included from /usr/include/qt4/QtCore/qvariant.h:48:0,
>                  from /usr/include/qt4/QtCore/qlocale.h:45,
>                  from /usr/include/qt4/QtCore/QtCore:20,
>                  from /usr/include/qt4/QtTest/QtTest:3,
>                  from testquazipfileinfo.cpp:10:
> /usr/include/qt4/QtCore/qmetatype.h: In instantiation of ‘static int QMetaTypeId2<T>::qt_metatype_id() [with T = QList<int>]’:
> /usr/include/qt4/QtCore/qmetatype.h:230:43:   required from ‘int qMetaTypeId(T*) [with T = QList<int>]’
> testquazipfileinfo.cpp:191:5:   required from here
> /usr/include/qt4/QtCore/qmetatype.h:169:79: error: ‘qt_metatype_id’ is not a member of ‘QMetaTypeId<QList<int> >’
>      static inline int qt_metatype_id() { return QMetaTypeId<T>::qt_metatype_id(); }
>                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
> /usr/include/qt4/QtCore/qmetatype.h: In instantiation of ‘static int QMetaTypeId2<T>::qt_metatype_id() [with T = QHash<short unsigned int, QList<QByteArray> >]’:
> /usr/include/qt4/QtCore/qmetatype.h:230:43:   required from ‘int qMetaTypeId(T*) [with T = QHash<short unsigned int, QList<QByteArray> >]’
> testquazipfileinfo.cpp:325:5:   required from here
> /usr/include/qt4/QtCore/qmetatype.h:169:79: error: ‘qt_metatype_id’ is not a member of ‘QMetaTypeId<QHash<short unsigned int, QList<QByteArray> > >’
>

--- a/qztest/testquazipfileinfo.cpp
+++ b/qztest/testquazipfileinfo.cpp
@@ -14,6 +14,11 @@
 #include "quazip/quazipfileinfo.h"
 #include "quazip/quazipnewinfo.h"
 
+#if QT_VERSION < 0x050000
+Q_DECLARE_METATYPE(QList<qint32>);
+Q_DECLARE_METATYPE(QuaExtraFieldHash);
+#endif
+
 TestQuaZipFileInfo::TestQuaZipFileInfo(QObject *parent) :
     QObject(parent)
 {

